### PR TITLE
Fix the analyze failure error message

### LINKED ISSUE #20043: [BUG] Error message when "update public API" script is not run is incredibly unhelpful.
**Describe the bug**
The error message when the public API script has not been run is not incredibly helpful.

**Expected behavior**
You get a message indicating that you need to run the Update Public API script.

**Actual behavior (include Exception or Stack Trace)**
You get a message telling you to run GenerateCode:

```
2021-04-01T21:22:33.5221619Z      {
2021-04-01T21:22:34.1892722Z ##[error]Generated code is not up to date.
2021-04-01T21:22:34.1905162Z 
2021-04-01T21:22:34.1905889Z error: Generated code is not up to date.
2021-04-01T21:22:34.1906326Z 
2021-04-01T21:22:34.1907628Z run 'dotnet build /t:GenerateCode' to update the generated code.
2021-04-01T21:22:34.1908177Z 
2021-04-01T21:22:34.1908633Z Summary:
2021-04-01T21:22:34.1908925Z 
2021-04-01T21:22:34.1928716Z    1 error(s)
2021-04-01T21:22:34.1932280Z 
2021-04-01T21:22:34.1936850Z error : Generated code is not up to date.
```

https://dev.azure.com/azure-sdk/public/_build/results?buildId=818797&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=72c41882-9d21-5580-4869-7f3ea1bd9c0a

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Change the shape of your API.
2. Start a pull request.

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.2.0] 
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8]
 - IDE and version : [e.g. Visual Studio 16.3]

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/20043

### CODE REVIEW

**Comment by pakrym** [comment]
@jsquire PTAL

**Comment by jsquire** [comment]
Clever, introducing the failure to demonstrate the new experience:

image

**Review by jsquire — APPROVED** [review]
LGTM;  the new display is much more clear and explicit.   Thanks, @pakrym!

File: eng/scripts/CodeChecks.ps1 [hunk 1]
```diff
@@ -98,11 +98,14 @@ try {
...
-                "run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#public-api-additions) `n" +
-                "run 'dotnet build /t:GenerateCode' to update the generated code."
+            LogError `
+"Generated code is not up to date.`
+    You may need to rebase on the latest master, `
+    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-sample-snippets), `
+    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#public-api-additions). `
+    run 'dotnet build /t:GenerateCode' to update the generated code.`
+    `
+To reproduce this error localy run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory $ServiceDirectory'."
```
Comment (weshaggard) [inline eng/scripts/CodeChecks.ps1:108]: localy -> locally

**Pull request merged**